### PR TITLE
487 update event template data model

### LINKED ISSUE #487: Update EventTemplate data model due to changed requerements
### Overview
Update [EventTemplate ](https://github.com/hackforla/VRMS/blob/development/backend/models/eventTemplate.model.js) model due to the [doc](https://docs.google.com/document/d/16VvC0fbaXsvEQkQaVSj8dG3wFEHgErbjJ6H3akSUoVM/edit#heading=h.aui7osumr40e).

### Action items

- [ ] Update the model
- [ ] Fix tests

### PR DESCRIPTION
**Pull request by akibrhast**
Closes #487 

This PR updates the event template model and the event template model test as requested in #487 following the specifications in this [doc](https://docs.google.com/document/d/16VvC0fbaXsvEQkQaVSj8dG3wFEHgErbjJ6H3akSUoVM/edit#heading=h.aui7osumr40e)

### CODE REVIEW

**Comment by akibrhast** [comment]
~~@alex-anakin Please don't merge this even if everything does look good~~

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: I'm not sure that making variables for all of these really buys you anything over doing it inline.  It seems nice and tidy, but it's also an extra line of code that you have to search for if you are debugging, or want to make changes.  It's a style thing, and I don't really have a problem with it, but if you are going to make variables for these, then I'd recommend making it consistent and doing a variable for 'type' as well.

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: Typos on "fridaya, saturdaya' in the comment.

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: Typo 'Ative' in comment

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: This is my super-nitpickyest comment of all, but the number of spaces after the ":" on your code isn't consistent -- sometimes it's a single space, sometimes it's double.  It makes my eye keep going back to it because inconsistencies make my eye think there's a problem when I'm scanning code.  Obviously, not an functional problem, but it would make me smile if they were consistent.

File: backend/models/eventTemplate.test.js
Comment (drubgrubby) [inline backend/models/eventTemplate.test.js]: Is there a reason that "locationZone" is commented out, but still here?

**Review by drubgrubby — CHANGES_REQUESTED** [review]
@akibrhast - This looks good to me.  These are mostly nit-picky style comments.  The only one that seems important (to me) is to make the variables consistent and either use them for everything, or not at all.  

Nice work!

File: backend/models/eventTemplate.model.js
Comment (akibrhast) [reply backend/models/eventTemplate.model.js]: ref: should possibly be LocationZone and NOT timeZone

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: I believe IDs are alphanumeric ( ex: 5eb9e2de3409ffef8194ffde) so I think this has to be type "string". ( That seems to be what other models use).

**Review by drubgrubby — DISMISSED** [review]
@akibrhast - You might want to confirm my comment about the ID type, but I'm pretty sure that's correct.  Otherwise looks good.  Thanks for making the quotation marks consistent.

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js]: @alex-anakin -> Just curious, is "project" different from "belongsToProjectID"?

File: backend/models/eventTemplate.model.js
Comment (drubgrubby) [inline backend/models/eventTemplate.model.js:36]: Should this also be type "mongoose.Schema.Types.ObjectId"?

**Review by drubgrubby — COMMENTED** [review]
Both of the comments here have to do with the type for the various ID fields. Probably wait for Alex to weigh in so we're sure that we're doing this in the right direction.

**Comment by akibrhast** [comment]
Hey @alex-anakin 
I did not really understand your explanation on the difference between the type string vs type objectID
and when we should use one vs the another. Could you please clarify on the reason behind why 
- `templateID`
- `eventManagerID`
- `belongsToProjectID`

should be of `type:string` and NOT `type: mongoose.Schema.Types.ObjectId`

A string seems to lax a check on something that is a type of ID and follows certain constraints. Plus would it not be preferable to use built in validators for a mongooseID and not have to write our own validators?

### Possible Resources
- https://stackoverflow.com/questions/13850819/can-i-determine-if-a-string-is-a-mongodb-objectid/29231016#29231016
- https://mongoosejs.com/docs/schematypes.html#objectids
- https://docs.mongodb.com/manual/reference/method/ObjectId/
- https://github.com/Automattic/mongoose/issues/1959

**Comment by peteplass** [comment]
Talked about in meeting, Alex to review

**Comment by akibrhast** [comment]
Update from @alex-anakin via slack

> in two words, if you have ObjectIds you are always able to fetch additional data. That is the "freedom" for me. Type checking is not important for ids because even if the type is correct but the value is not you won't fetch data

` if you have ObjectIds you are always able to fetch additional data.` ... So why is this bad?

> Somebody can add a "populate" method for fetching data when it's not needed. API will return extra data every time. In some cases, there will be performance issues. On other side, frontend data handling can be more complex when API response contains redundant data. - @alex-anakin 

As such based on that answer I am changing the id instances to Strings

**Comment by akibrhast** [comment]
## Progress
None since my last comment

## Blockers
busy with hfla/website

## Availability
pretty free this week

## ETA
by friday

File: backend/models/eventTemplate.model.js
Comment (akibrhast) [reply backend/models/eventTemplate.model.js:36]: been changed to string

**Comment by akibrhast** [comment]
@alex-anakin this is ready for review again

File: backend/models/eventTemplate.model.js
Comment (alex-anakin) [reply backend/models/eventTemplate.model.js]: We don't need a reference to the project here. 
But field `belongsToProjectID` emulates a foreign key.
@akibrhast please remove the `project` field from the model.

**Review by alex-anakin — CHANGES_REQUESTED** [review]
Please remove the `project` field from the model (see my comment to the code).

**Review by alex-anakin — APPROVED** [review]

**Pull request merged**
